test/lib/ufe/testComboCmd.py

```diff
@@ -395,3 +430,249 @@ def ball35Scale():
+
+        cmds.file(new=True, force=True)
+        mayaSphere = cmds.polySphere()[0]
+        
+        cmds.rotate(0, 0, -45, r=True, os=True, fo=True)
+        cmds.scale(4, 3, 2, r=True)
+        cmds.move(-2, -3, -4, "pSphere1.rotatePivot", r=True)
+        cmds.move(7, 6, 5, "pSphere1.scalePivot", r=True)
+
+        # Export out, reference back in using proxy shape.
+        usdFilePath = os.path.abspath('UsdExportMayaXformStack.usda')
+        cmds.mayaUSDExport(file=usdFilePath)
+
+        # Reference it back in.
+        proxyShape = cmds.createNode('mayaUsdProxyShape')
+        cmds.setAttr('mayaUsdProxyShape1.filePath', usdFilePath, type='string')
+
+        # MAYA-101766: awkward plug access for non-interactive stage loading.
+        outStageData = nameToPlug('mayaUsdProxyShape1.outStageData')
+        outStageData.asMDataHandle()
+
+        proxyShapeMayaPath = cmds.ls(proxyShape, long=True)[0]
+        proxyShapePathSegment = mayaUtils.createUfePathSegment(
+            proxyShapeMayaPath)
+        
+        spherePathSegment = usdUtils.createUfePathSegment('/pSphere1')
+        spherePath = ufe.Path([proxyShapePathSegment, spherePathSegment])
+        sphereItem = ufe.Hierarchy.createItem(spherePath)
+        usdSphereT3d = ufe.Transform3d.transform3d(sphereItem)
+        
+        # If the Transform3d interface can't handle rotate or scale pivot
+        # compensation, skip this test.
+        if usdSphereT3d.translateRotatePivotCmd() is None or \
+           usdSphereT3d.translateScalePivotCmd() is None:
+            raise unittest.SkipTest("Rotate or scale pivot compensation unsupported.")
+
+        # Maya object and its exported USD object twin should have the
+        # same pivots and pivot compensations.
+        checkPivotsAndCompensations(self, "pSphere1", usdSphereT3d)
+
+        sn = ufe.GlobalSelection.get()
+        sn.clear()
+        sn.append(sphereItem)
+
+        # Move only the rotate pivot.
+        cmds.move(-1, -2, -3, r=True, urp=True)
+        cmds.move(-1, -2, -3, "pSphere1.rotatePivot", r=True)        
+
+        checkPivotsAndCompensations(self, "pSphere1", usdSphereT3d)
+
+        # Move only the scale pivot.
+        cmds.move(-4, -3, -2, r=True, usp=True)
+        cmds.move(-4, -3, -2, "pSphere1.scalePivot", r=True)
+
+        checkPivotsAndCompensations(self, "pSphere1", usdSphereT3d)
+
+    # Name test such that it runs last.  Otherwise, it runs before 
+    # testRotateScalePivotCompensation(), and causes it to fail.  To be 
+    # investigated --- MAYA-108067.
+    @unittest.skipUnless(ufeUtils.ufeFeatureSetVersion() >= 2, 'Fallback transform op handling only available in UFE v2 or greater.')
+    def testZFallback(self):
+        '''Transformable not handled by standard Transform3d handlers must be
+    handled by fallback handler.'''
+
+        mayaUtils.openTestScene("xformOpFallback", "fallbackTest.ma")
+
+        # We have three objects in the scene, one Maya, one USD with a Maya
```

Comment:
Excellent tests. One addition that would be useful to make is for testing fallback when editing pivots on matrix stacks.